ocaml/xapi/create_storage.ml

```diff
@@ -75,13 +75,44 @@ let maybe_create_pbd rpc session_id sr device_config me =
   then Client.PBD.create ~rpc ~session_id ~host:me ~sR:sr ~device_config ~other_config:[]
   else List.hd pbds (* Otherwise, return the current one *)
 
-let create_storage (me: API.ref_host) rpc session_id __context : unit =
+let maybe_remove_tools_sr rpc session_id __context =
+  let srs = Db.SR.get_all ~__context in
+  let tools_srs =
+    List.filter (fun self -> Db.SR.get_is_tools_sr ~__context ~self) srs in
+  let old_srs =
+    List.filter (fun self ->
+        let other_config = Db.SR.get_other_config ~__context ~self in
+        (Db.SR.get_is_tools_sr ~__context ~self = false) &&
+        ((List.mem_assoc Xapi_globs.tools_sr_tag other_config) ||
+         (List.mem_assoc Xapi_globs.xensource_internal other_config))
+      ) srs in
+  let unplug_and_maybe_destroy sr =
+    (* Unplug me *)
+    let pbd, _ = Sm.get_my_pbd_for_sr __context sr in
+    Db.PBD.set_currently_attached ~__context ~self:pbd ~value:false;
```

Comment:
I assume this is to prevent more uses of the tools SR, should the below forget fail because the SR is in-use (e.g. by a plugged in cd-rom device)

Reply:
The unplug means that this host is ready for the Tools SR to be removed. SR.forget will only work if all PBDs are unplugged.